var/spack/repos/builtin/packages/octopus/package.py

```diff
@@ -316,39 +316,25 @@ def configure_args(self):
 
     @run_after("install")
     @on_package_attributes(run_tests=True)
-    def smoke_tests_after_install(self):
+    def benchmark_tests_after_install(self):
         """Function stub to run tests after install if desired
         (for example through `spack install --test=root octopus`)
         """
-        self.smoke_tests()
+        self.test_version()
+        self.test_example()
+        self.test_he()
 
-    def test(self):
-        """Entry point for smoke tests run through `spack test run octopus`."""
-        self.smoke_tests()
-
-    def smoke_tests(self):
-        """Actual smoke tests for Octopus."""
-        #
-        # run "octopus --version"
-        #
-        exe = join_path(self.spec.prefix.bin, "octopus")
-        options = ["--version"]
-        purpose = "Check octopus can execute (--version)"
+    def test_version(self):
+        """Check octopus can execute (--version)"""
         # Example output:
         #
         # spack-v0.17.2$ octopus --version
         # octopus 11.3 (git commit )
         expected = ["octopus "]
 
-        self.run_test(
-            exe,
-            options=options,
-            expected=expected,
-            status=[0],
-            installed=False,
-            purpose=purpose,
-            skip_missing=False,
-        )
+        exe = which(self.spec.prefix.bin, "octopus")
+        out = exe("--version", output=str.split, error=str.split)
+        check_outputs(expected, out)
```

Comment:
There's only one expected output so an `assert` is more appropriate (and less overhead) here.

```suggestion
        assert "octopus " in out
```